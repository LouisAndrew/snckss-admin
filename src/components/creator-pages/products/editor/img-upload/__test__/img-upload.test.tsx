import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import ImgUploader from '..'

describe('img uploader element', () => {
        const mockImgUrls: string[] = [
                'https://res.cloudinary.com/dsvdffre0/image/upload/v1594991459/wzv9c2057bbqh6ht8kwi.jpg',
                'https://res.cloudinary.com/dsvdffre0/image/upload/v1594991470/oyjs4mo3xwksv7w7dvsf.png',
        ]
        afterEach(cleanup)
        const el: React.ReactElement = <ImgUploader />
        const elWithImgs: React.ReactElement = (
                <ImgUploader allImgs={mockImgUrls} />
        )

        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(el, div)
        })

        it('should render when a button clicked correctly', () => {
                const { getByRole } = render(el)
                const button: HTMLButtonElement = getByRole(
                        'button'
                ) as HTMLButtonElement

                fireEvent.click(button)
                const elements: NodeListOf<Element> = document.querySelectorAll(
                        '.img-upload'
                )
                elements.forEach((el) => {
                        expect(el).toBeInTheDocument()
                })
        })

        it('should render when a button clicked multiple times correctly', () => {
                const { getByRole } = render(el)
                const button: HTMLButtonElement = getByRole(
                        'button'
                ) as HTMLButtonElement

                const timesClicked: number = 4
                for (let i = 0; i < timesClicked; i++) {
                        fireEvent.click(button)
                }

                const elements: NodeListOf<Element> = document.querySelectorAll(
                        '.img-upload'
                )
                expect(elements.length).toBe(timesClicked)
        })

        // TODO: test img elements.
        it('should render correctly when an array of urls is being passed as props', () => {
                const { getAllByRole } = render(elWithImgs)
                const imgs: HTMLImageElement[] = getAllByRole(
                        'img'
                ) as HTMLImageElement[]

                imgs.forEach((img, i) => {
                        expect(img.src).toBe(mockImgUrls[i])
                })
        })

        it('matches snapshot', () => {
                const tree = renderer.create(el).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
