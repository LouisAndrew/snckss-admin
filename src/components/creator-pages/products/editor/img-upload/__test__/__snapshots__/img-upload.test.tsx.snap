// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`img uploader element matches snapshot 1`] = `
<div
  className="upload"
>
  <label
    className=""
  >
    Product images
  </label>
  <button
    aria-label={null}
    className="btn btn-dark"
    onClick={[Function]}
    type="button"
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
  </button>
</div>
`;
